sigopt_sklearn/search.py

```diff
@@ -274,7 +276,8 @@ def _fit(self, X, y, parameter_iterable=None):
         pre_dispatch = self.pre_dispatch
 
         # setup SigOpt experiment and run optimization
-        self._create_sigopt_exp(self.sigopt_connection)
+        if self.experiment is None:
+            self.experiment = self._create_sigopt_exp(self.sigopt_connection)
```

Comment:
can we make this public? otherwise users have to instantiate their own connection and make their own compatible experiment in order to use this feature

if we make this public and then defer creating the `experiment` object until `fit` is called then we can support the following workflows:

```
SigOptSearchCV(...).fit(X, y)
```

```
c = SigOptSearchCV(...)
e = c.create_experiment()  # returns an ID
c.fit(X, y, e)
```

```
EXPERIMENT_ID = 5
c = SigOptSearchCV(...)
c.fit(X, y, EXPERIMENT_ID)
```

wdyt @iandewancker ?